### PR TITLE
Gracefully handle empty CSV

### PR DESCRIPTION
**Pull request by raghumdani**
#205

### Testing

Unit tests added.

### CODE REVIEW

File: deltacat/tests/utils/data/non_empty_valid.csv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Y	1754-08-30T22:43:41
+""	2022-08-10T22:43:21
+N	2022-08-10T22:43:21
```
Comment (yankevn) [inline deltacat/tests/utils/data/non_empty_valid.csv:3]: Should we also test `null` (truly empty values), timestamps that are far in the future, and timestamps with decimal values?

File: deltacat/utils/pyarrow.py [hunk 2]
```diff
@@ -33,15 +33,74 @@
...
+    for column_name in columns:
+        index = schema.get_field_index(column_name)
+
+        if index != -1:
+            target_field = schema.field(index)
+            target_schema_fields.append(target_field)
+
+    target_schema = pa.schema(target_schema_fields, metadata=schema.metadata)
+
+    return target_schema
```
Comment (yankevn) [inline deltacat/utils/pyarrow.py:57]: So what exactly is this function doing? It might be good to add a docstring. Why would we want to remove columns from the schema?

File: deltacat/tests/utils/data/non_empty_valid.csv (hunk 1 above)
Comment (raghumdani) [reply deltacat/tests/utils/data/non_empty_valid.csv:3]: Formatter used by DeltaCAT is cleaning it up. Hence, didn't include it, but it pyarrow related functionality.

File: deltacat/utils/pyarrow.py (hunk 2 above)
Comment (raghumdani) [reply deltacat/utils/pyarrow.py:57]: Creates a new schema that adheres to the given column name ordering and types.

File: deltacat/tests/utils/data/non_empty_valid.csv (hunk 1 above)
Comment (yankevn) [reply deltacat/tests/utils/data/non_empty_valid.csv:3]: Is that just for nulls? Is that only if the first field/second field is null, or it doesn't matter and the formatter always complains if any field is null? Could we still test timestamps like `9999-08-10T22:43:21.123456`

Comment (raghumdani) [reply deltacat/tests/utils/data/non_empty_valid.csv:3]: Sure, added year 9999.

File: deltacat/utils/pyarrow.py [hunk 3]
```diff
@@ -33,15 +33,74 @@
...
+
+
+def pyarrow_read_csv(*args, **kwargs) -> pa.Table:
+    try:
+        kwargs = sanitize_kwargs_by_supported_kwargs(
+            ["read_options", "parse_options", "convert_options", "memory_pool"], kwargs
+        )
+        return pacsv.read_csv(*args, **kwargs)
+    except pa.lib.ArrowInvalid as e:
+        if e.__str__() == "Empty CSV file":
```
Comment (rkenmi) [inline deltacat/utils/pyarrow.py:67]: This is a workaround specific to quirks of our own internal data catalog, which ideally shouldn't be producing empty CSVs anyway. As a general solution, perhaps its better to provide an option to disable/enable this functionality if desired?

Comment (raghumdani) [reply deltacat/utils/pyarrow.py:67]: Good point.

**Review by rkenmi — APPROVED** [review]
